src/garage/_dtypes.py

```diff
@@ -167,3 +166,128 @@ def __new__(cls, env_spec, observations, actions, rewards, terminals,
         return super().__new__(TrajectoryBatch, env_spec, observations,
                                actions, rewards, terminals, env_infos,
                                agent_infos, lengths)
+
+
+class Sample(
+        collections.namedtuple('Sample', [
+            'env_spec',
+            'observation',
+            'action',
+            'reward',
+            'next_observation',
+            'terminal',
+            'env_info',
+            'agent_info',
+        ])):
+    # pylint: disable=missing-return-doc, missing-return-type-doc, missing-param-doc, missing-type-doc  # noqa: E501
+    r"""A tuple representing a single sample.
+
+    A :class:`Sample` represents a single sample when an agent interacts with
+        an environment.
+
+    Attributes:
+        env_spec (garage.envs.EnvSpec): Specification for the environment from
+            which this data was sampled.
+        observation (numpy.ndarray): A numpy array of shape
+            :math:`(O^*)` containing the observation for the current time step
+            in the environment. These must conform to
+            :obj:`env_spec.observation_space`.
+            actions (numpy.ndarray): A numpy array of shape
+            :math:`(A^*)` containing the action for the current time step.
+            These must conform to :obj:`env_spec.action_space`.
+        reward (float): A float representing the reward for taking
+            the action given the observation, at the current time step.
+        terminals (bool): The termination signal for the current time step.
+        env_info (dict): A dict arbitrary environment state
+            information.
+        agent_infos (numpy.ndarray): A dict of arbitrary agent
```

Comment:
`agent_info`? No need  for this to be plural, since there's only one.

Reply:
resolved, thanks